src/tests/modules/bson.rs

```diff
@@ -143,6 +143,15 @@ fn from_impls() {
         Bson::Document(doc! {"a": "b"})
     );
 
+    // Optionals
+    assert_eq!(Bson::from(Some(4)), Bson::Int32(4));
+    assert_eq!(
+        Bson::from(Some(String::from("data"))),
+        Bson::String(String::from("data"))
+    );
+    assert_eq!(Bson::from(None::<i32>), Bson::Null);
+    assert_eq!(Bson::from(None::<String>), Bson::Null);
```

Comment:
Can we also add a test here to ensure that this works properly in the `doc!` macro? Something along the lines of the code in the ticket description.

Reply:
I added tests for the `Some` and `None` cases. Let me know what you think.